exports/dataset.cpp

```diff
@@ -684,21 +684,49 @@ PYBIND11_MODULE(dataset, m) {
                     "variable. Raises an exception of the variable is "
                     "not 0-dimensional.")
       .def(py::self + py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self + float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self + double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self - py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self - float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self - double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self * py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self * float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self * double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self / py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self / float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self / double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self += py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self += float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self += double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self -= py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self -= float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self -= double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self *= py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self *= float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self *= double(), py::call_guard<py::gil_scoped_release>())
       .def(py::self /= py::self, py::call_guard<py::gil_scoped_release>())
-      .def(py::self /= float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self /= double(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self == py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self != py::self, py::call_guard<py::gil_scoped_release>())
+      .def("__eq__", [](Variable &a, VariableSlice &b) { return a == b; },
+           py::is_operator())
+      .def("__ne__", [](Variable &a, VariableSlice &b) { return a != b; },
+           py::is_operator())
+      .def("__add__", [](Variable &a, VariableSlice &b) { return a + b; },
```

Comment:
I think this would be handled by `py::self + py::self` line 689. `VariableSlice` is implicitly convertible to `Variable`. Am I missing something?

Reply:
I think this is prioritized instead of implicit conversion (i think it only uses implicit conversion if it can't find a specified method for the operation). If i change that `__add__` to be:
```C++
.def("__add__", [](Variable &a, VariableSlice &b) { return 999.0; },
           py::is_operator())
```
and then do
```Python
d1 = Dataset()
N = 10
d1[Coord.Tof] = ([Dim.Tof], np.arange(N).astype(np.float64))
d1[Data.Value, "A"] = ([Dim.Tof], np.arange(N, dtype=np.float64))
v1 = d1[Data.Value, "A"] # v1 is a VariableSlice
v2 = Variable(Data.Value, [], ()) # v2 is a Variable
v3 = v2 + v1
v3
```
i get `999.0` for `v3`, so it seems to be using that function.

Reply:
@OwenArnold Yes it would be handled by implicit conversion, but that involves an expensive copy operation, so we want to avoid that. Therefore, this overload is necessary.